site/cds_rdm/tasks.py

```diff
@@ -0,0 +1,113 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2023 CERN.
+#
+# Invenio is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Celery tasks for cds."""
+import requests
+from celery import shared_task
+from flask import current_app
+from invenio_oauthclient.handlers.base import create_or_update_groups
+
+from cds_rdm.errors import RequestError
+
+
+@shared_task(
+    bind=True, max_retries=6, default_retry_delay=60 * 10
+)  # Retry every 10 min for 1 hour
+def synch_groups(self):
+    """Synchronizes groups in CDS."""
+    token_url = (
+        f"{current_app.config['CERN_KEYCLOAK_URL']}auth/realms/cern/api-access/token"
+    )
+    token_data = {
+        "grant_type": "client_credentials",
+        "client_id": current_app.config["CERN_APP_CREDENTIALS"]["consumer_key"],
+        "client_secret": current_app.config["CERN_APP_CREDENTIALS"]["consumer_secret"],
+        "audience": "authorization-service-api",  # This is the target api of the token
+    }
+    try:
+        token_response = requests.post(url=token_url, data=token_data)
+    except Exception as e:
+        while self.request.retries < self.max_retries:
+            self.retry()
+        raise RequestError(token_url, str(e))
+
+    if not token_response.ok:
+        while self.request.retries < self.max_retries:
+            self.retry()
+        raise RequestError(
+            token_url,
+            f"Request failed with status code {token_response.status_code} {token_response.reason}.",
+        )
+
+    token = token_response.json()["access_token"]
+    offset = 0
+    limit = 1000
+    groups_headers = {
+        "Authorization": f"Bearer {token}",
+        "accept": "text/plain",
+    }
+
+    # We do this to get the total amount of entries, to be able to create as many celery tasks as required
+    groups_url = f"{current_app.config['CDS_AUTHORIZATION_SERVICE_API']}{current_app.config['CDS_AUTHORIZATION_SERVICE_API_GROUP_URL']}?offset=0&limit=1"
```

Comment:
improvement: move the url parameters to the CDS_AUTHORIZATION_SERVICE_API_GROUP_URL